test/test_other.py

```diff
@@ -2691,35 +2672,53 @@ def test_dwarf_sourcemap_names(self):
       ([], False, False, False),
       (['-g0'], False, False, False),
       (['-g1'], False, False, False),
+      (['-g1', '-O2'], False, False, False),
+      (['--minify=0'], False, False, False),
       # last numeric g flag "wins", so g0 overrides -g
       (['-g', '-g0'], False, False, False),
       (['-g2'], False, False, True),
       (['-gline-tables-only'], True, False, True),
       (['--profiling'], False, False, True),
       (['--profiling-funcs'], False, False, True),
+      (['-O2', '--profiling-funcs'], False, False, True),
       (['-g'], True, False, True),
       (['-g3'], True, False, True),
-      (['-O2', '-g'], True, False, True),
+      (['-O1', '-g'], True, False, True),
+      (['-O3', '-g'], True, False, True),
       (['-gsplit-dwarf'], True, False, True),
+      # TODO: It seems odd that -gsource-map leaves behind a name section. Should it?
       (['-gsource-map'], False, True, True),
-      (['-g0', '-gsource-map'], False, True, True),
+      (['-g1', '-Oz', '-gsource-map'], False, True, True),
       # -g0 does not override -gsource-map but does remove name section. TODO: should it?
       (['-gsource-map', '-g0'], False, True, False),
+      (['--emit-symbol-map', '-gsource-map'], False, True, True),
+      (['--emit-symbol-map'], False, False, False),
+      (['--emit-symbol-map', '-Oz'], False, False, False),
```

Comment:
Is `--emit-symbol-map` relevant to any of name section, DWARF, or source map generation? Adding it doesn't seem to change anything in these three options.

Reply:
I see. Will add
```
# --emit-symbol-map should not affect the results
```
Above these three tests to be clear.